File: persistence/src/main/java/io/sparkled/persistence/playlist/impl/query/GetPlaylistSequencesByPlaylistIdQuery.java
package io.sparkled.persistence.playlist.impl.query;

import io.sparkled.model.entity.PlaylistSequence;
import io.sparkled.persistence.PersistenceQuery;
import io.sparkled.persistence.QueryFactory;

import java.util.List;

public class GetPlaylistSequencesByPlaylistIdQuery implements PersistenceQuery<List<PlaylistSequence>> {

    private final int playlistId;

    public GetPlaylistSequencesByPlaylistIdQuery(int playlistId) {
        this.playlistId = playlistId;
    }

    @Override
    public List<PlaylistSequence> perform(QueryFactory queryFactory) {
        return queryFactory
                .selectFrom(qPlaylistSequence)
                .where(qPlaylistSequence.playlistId.eq(playlistId))
                .orderBy(qPlaylistSequence.displayOrder.asc())
                .fetch();
    }
}

File: persistence/src/main/kotlin/io/sparkled/persistence/playlist/impl/query/GetPlaylistSequencesByPlaylistIdQuery.kt
package io.sparkled.persistence.playlist.impl.query

import io.sparkled.model.entity.PlaylistSequence
import io.sparkled.model.entity.QPlaylistSequence.Companion.playlistSequence
import io.sparkled.persistence.PersistenceQuery
import io.sparkled.persistence.QueryFactory

class GetPlaylistSequencesByPlaylistIdQuery(private val playlistId: Int) : PersistenceQuery<List<PlaylistSequence>> {

    override fun perform(queryFactory: QueryFactory): List<PlaylistSequence> {
        return queryFactory
            .selectFrom(playlistSequence)
            .where(playlistSequence.playlistId.eq(playlistId))
            .orderBy(playlistSequence.displayOrder.asc())
            .fetch()
    }
}
